pkg/networkservice/chains/endpoint/combine_test.go

```diff
@@ -83,14 +83,23 @@ func testCombine(t *testing.T, mechanism *networkservice.Mechanism) {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 	defer cancel()
 
+	var m networkservice.MonitorConnectionServer
 	e := endpoint.Combine(func(servers []networkservice.NetworkServiceServer) networkservice.NetworkServiceServer {
-		return mechanisms.NewServer(map[string]networkservice.NetworkServiceServer{
-			kernel.MECHANISM: servers[0],
-			memif.MECHANISM:  servers[1],
-		})
-	}, newTestEndpoint(ctx, kernel.MECHANISM), newTestEndpoint(ctx, memif.MECHANISM))
-
-	cc := startEndpoint(ctx, t, e)
+		return next.NewNetworkServiceServer(
+			begin.NewServer(),
+			metadata.NewServer(),
+			monitor.NewServer(ctx, &m),
```

Comment:
In most cases, we're using these elements inside endpoints that we're combining. So I'd prefer to rework the test to keep it similar to cmd-forwarder-vpp. See at https://github.com/networkservicemesh/cmd-forwarder-vpp/blob/main/internal/xconnectns/server.go#L92